### ISSUE TITLE
Bug? Gluetun + connected container bring down host (macOS) network.

### ISSUE THREAD

**Issue by snorkrat** [qdm12/gluetun#1138]
This is a strange one.  Going to give as much info as possible to help troubleshoot my issue.  It  could potentially be a Gluetun issue, or a general Docker one.

I use Gluetun (:latest with FastestVPN) only with a qbittorrent container (lsio container).  Running on macOS 12.5.1,  I always keep my Gluetun image and Docker up to date, never more than a week behind updates.

I have noticed over the past maybe 6 months or so that in the mornings my qbittorrent and gluetun containers would have stopped overnight on their own.  This was a mild frustration. and I kept meaning to dig in and see what was causing it.  Occasionally it would happen during the day.  I would try to access qbittorent's webui and it wouldn't load.  Upon checking the containers, they were both down.  Nothing useful (imo) was output in the logs, in fact I don't remember seeing anything other than "Exited (1)" in the log (which makes me think this is _maybe_ a Docker issue?).  I never got around to digging into the problem, but over the past maybe, 2-3 weeks it has been happening a lot.

The main issue now is that it has been **bringing the host (macOS) network down with it**.  Which only gets resolved by stopping and removing the gluetun and qbit containers, and by restarting docker desktop.  It has become unbearable to use, happening within minutes (10-15?) of the containers being up.

I noticed this a lot more after building a nice dashboard in Grafana to show metrics and syncing state of rclone.  I use rclone to sync my external HDDs to Google Drive.  I set a bandwidth limit in rclone to use max 20Mbps of my 100Mbps internet _upload_ speed.  When creating the dashboard I noticed that the speed would start at ~20Mbps and then drop to ~7Mbps.  When the drop occurred this is when the host network of macOS would be down.  I could still remote into the Mac (headless Mac mini), so the network wasn't _completely_ down but none of my other docker containers would be accessible, and services that are running directly on the host, eg: Plex would also be inaccessible - meaning it wasn't just the docker network that was down.  Again, stopping and removing gluetun and qbit containers, followed by a restart of Docker would bring back the host network.
Note, running rclone without gluetun and qbit would not affect the host network. And in fact it was only through building the rclone dashboard that I could easily tell when the network dropped, it would drop even without rclone running.

I have attached my docker-compose files.  One for my "main" stack, and one for the gluetun/qbit stack.  Note: I have taken away a lot of environment variables in the main stack.  Showing in case there is conflict with the network setup or other containers, but probably irrelevant anyway.

[Main Stack](https://pastebin.com/iWMMUWFB)
[Gluetun Stack](https://pastebin.com/93tN16fj)

I'm kind of pulling my hair out, and grasping at straws to try and find out why this is happening.  I know it has _something_ to do with running qbittorent connected to Gluetun, as it only happens when they are running.  But in my gut I feel like it is a Docker for Mac issue and something is broken in the networking side of things.

Hopefully somebody can shed some light on why this is happening, or at least point me in the right direction!

Thanks!

There is probably a stack of extra information above, but I wanted to give all info about when and how the issue occurs.  If it was just the containers going down I could live with it, but the fact that it brings the host network down is unbearable for me.

Edit: Fixed URL

**Comment by snorkrat** [comment]
Adding Gluetun logs after the issue just happened.  I notice these which I have never seen before

```
2022-09-08T09:39:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] write UDP [EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:59:04+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)

```
Full log below:
```
Running version latest built on 2022-09-06T12:22:01.166Z (commit 7506625)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2022-09-08T09:21:54+01:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.10
2022-09-08T09:21:54+01:00 INFO [routing] local ethernet link found: eth0
2022-09-08T09:21:54+01:00 INFO [routing] local ipnet found: 172.18.0.0/16
2022-09-08T09:21:54+01:00 INFO [firewall] enabling...
2022-09-08T09:21:54+01:00 INFO [firewall] enabled successfully
2022-09-08T09:21:54+01:00 INFO [storage] creating /gluetun/servers.json with 11713 hardcoded servers
2022-09-08T09:21:55+01:00 INFO Alpine version: 3.16.2
2022-09-08T09:21:55+01:00 INFO OpenVPN 2.4 version: 2.4.12
2022-09-08T09:21:55+01:00 INFO OpenVPN 2.5 version: 2.5.6
2022-09-08T09:21:55+01:00 INFO Unbound version: 1.15.0
2022-09-08T09:21:55+01:00 INFO IPtables version: v1.8.8
2022-09-08T09:21:55+01:00 INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: fastestvpn
|   |   └── Server selection settings:
|   |       ├── VPN type: openvpn
|   |       ├── Countries: uk
|   |       └── OpenVPN server selection settings:
|   |           └── Protocol: UDP
|   └── OpenVPN settings:
|       ├── OpenVPN version: 2.5
|       ├── User: [set]
|       ├── Password: [set]
|       ├── Network interface: tun0
|       ├── Run OpenVPN as: root
|       └── Verbosity level: 1
├── DNS settings:
|   ├── DNS server address to use: 127.0.0.1
|   ├── Keep existing nameserver(s): no
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       |       ├── 0.0.0.0/0
|       |       └── ::/0
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:7f00:1/104
|               ├── ::ffff:a00:0/104
|               ├── ::ffff:a9fe:0/112
|               ├── ::ffff:ac10:0/108
|               └── ::ffff:c0a8:0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Target address: cloudflare.com:443
|   ├── Read header timeout: 100ms
|   ├── Read timeout: 500ms
|   └── VPN wait durations:
|       ├── Initial duration: 6s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 501
|   ├── Process GID: 20
|   └── Timezone: Europe/London
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2022-09-08T09:21:55+01:00 INFO IPv6 is not supported
2022-09-08T09:21:55+01:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.10
2022-09-08T09:21:55+01:00 INFO [routing] adding route for 0.0.0.0/0
2022-09-08T09:21:55+01:00 INFO [firewall] setting allowed subnets...
2022-09-08T09:21:55+01:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.10
2022-09-08T09:21:55+01:00 INFO TUN device is not available: open /dev/net/tun: no such file or directory; creating it...
2022-09-08T09:21:55+01:00 INFO [pprof] http server listening on [::]:6060
2022-09-08T09:21:55+01:00 INFO [http server] http server listening on [::]:8000
2022-09-08T09:21:55+01:00 INFO [dns over tls] using plaintext DNS at address 1.1.1.1
2022-09-08T09:21:55+01:00 INFO [healthcheck] listening on 127.0.0.1:9999
2022-09-08T09:21:55+01:00 INFO [firewall] allowing VPN connection...
2022-09-08T09:21:55+01:00 WARN [openvpn] Compression for receiving enabled. Compression has been used in the past to break encryption. Sent packets are not compressed unless "allow-compression yes" is also set.
2022-09-08T09:21:55+01:00 INFO [openvpn] OpenVPN 2.5.6 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 17 2022
2022-09-08T09:21:55+01:00 INFO [openvpn] library versions: OpenSSL 1.1.1q  5 Jul 2022, LZO 2.10
2022-09-08T09:21:55+01:00 WARN [openvpn] No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.
2022-09-08T09:21:55+01:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]5.226.139.148:4443
2022-09-08T09:21:55+01:00 INFO [openvpn] UDP link local: (not bound)
2022-09-08T09:21:55+01:00 INFO [openvpn] UDP link remote: [AF_INET]5.226.139.148:4443
2022-09-08T09:21:55+01:00 WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1570'
2022-09-08T09:21:55+01:00 WARN [openvpn] 'tun-mtu' is used inconsistently, local='tun-mtu 1532', remote='tun-mtu 1500'
2022-09-08T09:21:55+01:00 INFO [openvpn] [FastestVPN] Peer Connection Initiated with [AF_INET]5.226.139.148:4443
2022-09-08T09:21:56+01:00 ERROR [openvpn] Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:5: block-outside-dns (2.5.6)
2022-09-08T09:21:56+01:00 INFO [openvpn] setsockopt TCP_NODELAY=1 failed
2022-09-08T09:21:56+01:00 INFO [openvpn] TUN/TAP device tun0 opened
2022-09-08T09:21:56+01:00 INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2022-09-08T09:21:56+01:00 INFO [openvpn] /sbin/ip link set dev tun0 up
2022-09-08T09:21:56+01:00 INFO [openvpn] /sbin/ip addr add dev tun0 10.16.0.6/16
2022-09-08T09:21:56+01:00 INFO [openvpn] UID set to nonrootuser
2022-09-08T09:21:56+01:00 INFO [openvpn] Initialization Sequence Completed
2022-09-08T09:21:56+01:00 INFO [dns over tls] downloading DNS over TLS cryptographic files
2022-09-08T09:21:57+01:00 INFO [healthcheck] healthy!
2022-09-08T09:21:57+01:00 INFO [dns over tls] downloading hostnames and IP block lists
2022-09-08T09:22:00+01:00 INFO [dns over tls] init module 0: validator
2022-09-08T09:22:00+01:00 INFO [dns over tls] init module 1: iterator
2022-09-08T09:22:00+01:00 INFO [dns over tls] start of service (unbound 1.15.0).
2022-09-08T09:22:00+01:00 INFO [dns over tls] generate keytag query _ta-4a5c-4f66. NULL IN
2022-09-08T09:22:00+01:00 INFO [dns over tls] ready
2022-09-08T09:22:00+01:00 INFO [ip getter] Public IP address is 5.226.139.148 (United Kingdom, England, London)
2022-09-08T09:22:01+01:00 INFO [vpn] You are running on the bleeding edge of latest!
2022-09-08T09:39:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] write UDP [EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:49:00+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452|EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T09:59:04+01:00 INFO [openvpn] read UDP [EMSGSIZE Path-MTU=1452]: Message too large (code=90)
2022-09-08T10:02:21+01:00 INFO [healthcheck] unhealthy: cannot dial: dial tcp4 104.16.132.229:443: i/o timeout
2022-09-08T10:02:29+01:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN
2022-09-08T10:02:29+01:00 INFO [vpn] stopping
2022-09-08T10:02:46+01:00 INFO [vpn] starting
2022-09-08T10:02:46+01:00 INFO [firewall] allowing VPN connection...
2022-09-08T10:02:46+01:00 WARN [openvpn] Compression for receiving enabled. Compression has been used in the past to break encryption. Sent packets are not compressed unless "allow-compression yes" is also set.
2022-09-08T10:02:46+01:00 INFO [openvpn] OpenVPN 2.5.6 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 17 2022
2022-09-08T10:02:46+01:00 INFO [openvpn] library versions: OpenSSL 1.1.1q  5 Jul 2022, LZO 2.10
2022-09-08T10:02:46+01:00 WARN [openvpn] No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.
2022-09-08T10:02:46+01:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]83.229.75.27:4443
2022-09-08T10:02:46+01:00 INFO [openvpn] UDP link local: (not bound)
2022-09-08T10:02:46+01:00 INFO [openvpn] UDP link remote: [AF_INET]83.229.75.27:4443
2022-09-08T10:02:57+01:00 INFO [healthcheck] program has been unhealthy for 11s: restarting VPN
2022-09-08T10:02:57+01:00 INFO [vpn] stopping
2022-09-08T10:02:57+01:00 INFO [vpn] starting
2022-09-08T10:03:20+01:00 INFO [firewall] allowing VPN connection...
2022-09-08T10:03:20+01:00 WARN [openvpn] Compression for receiving enabled. Compression has been used in the past to break encryption. Sent packets are not compressed unless "allow-compression yes" is also set.
2022-09-08T10:03:20+01:00 INFO [openvpn] OpenVPN 2.5.6 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 17 2022
2022-09-08T10:03:20+01:00 INFO [openvpn] library versions: OpenSSL 1.1.1q  5 Jul 2022, LZO 2.10
2022-09-08T10:03:20+01:00 WARN [openvpn] No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.
2022-09-08T10:03:20+01:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]5.226.139.148:4443
2022-09-08T10:03:20+01:00 INFO [openvpn] UDP link local: (not bound)
2022-09-08T10:03:20+01:00 INFO [openvpn] UDP link remote: [AF_INET]5.226.139.148:4443
2022-09-08T10:03:21+01:00 WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1570'
2022-09-08T10:03:21+01:00 WARN [openvpn] 'tun-mtu' is used inconsistently, local='tun-mtu 1532', remote='tun-mtu 1500'
2022-09-08T10:03:21+01:00 INFO [openvpn] [FastestVPN] Peer Connection Initiated with [AF_INET]5.226.139.148:4443
2022-09-08T10:03:22+01:00 ERROR [openvpn] Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:5: block-outside-dns (2.5.6)
2022-09-08T10:03:22+01:00 INFO [openvpn] setsockopt TCP_NODELAY=1 failed
2022-09-08T10:03:22+01:00 INFO [openvpn] TUN/TAP device tun0 opened
2022-09-08T10:03:22+01:00 INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2022-09-08T10:03:22+01:00 INFO [openvpn] /sbin/ip link set dev tun0 up
2022-09-08T10:03:22+01:00 INFO [openvpn] /sbin/ip addr add dev tun0 10.16.0.19/16
2022-09-08T10:03:22+01:00 ERROR [openvpn] RTNETLINK answers: File exists
2022-09-08T10:03:22+01:00 ERROR [openvpn] Linux route add command failed: external program exited with error status: 2
2022-09-08T10:03:22+01:00 INFO [openvpn] UID set to nonrootuser
2022-09-08T10:03:22+01:00 INFO [openvpn] Initialization Sequence Completed
2022-09-08T10:03:22+01:00 INFO [ip getter] Public IP address is 5.226.139.148 (United Kingdom, England, London)
2022-09-08T10:03:23+01:00 INFO [healthcheck] healthy!
2022-09-08T10:04:06+01:00 INFO [healthcheck] unhealthy: cannot dial: dial tcp4 104.16.133.229:443: i/o timeout
2022-09-08T10:04:11+01:00 INFO [healthcheck] healthy!
2022-09-08T10:04:45+01:00 INFO [healthcheck] unhealthy: cannot dial: dial tcp4 104.16.132.229:443: i/o timeout
2022-09-08T10:04:45+01:00 INFO [healthcheck] healthy!
2022-09-08T10:04:58+01:00 INFO [healthcheck] unhealthy: cannot dial: dial tcp4 104.16.132.229:443: i/o timeout
2022-09-08T10:04:59+01:00 INFO [healthcheck] healthy!
```

**Comment by snorkrat** [comment]
We can actually go ahead and close this down.  It was just coincidence and bad timing that made me think my gluetun stack was the culprit.  I ran my main stack on it's own and, while it was a little longer before my network dropped out, the exact same thing happened.
I feel it is to do with telegraf (on the host) sending a lot more metrics to influxdb/grfana on docker since I made my rclone dashboard.

Anyway, it's not a Gluetun issue so closing.
